check_yum.py

```diff
@@ -322,33 +322,25 @@ def get_all_updates(self):
         if self.verbosity >= 4:
             for section in output2:
                 print("\nSection:\n%s\n" % section)
-        if len(output2) > 2 or \
+        if len(output2) > 2 or (not self.no_cache_update and \
            not ("Setting up repositories" in output2[0] or \
                 "Loaded plugins: " in output2[0] or \
                 "Last metadata expiration check" in output2[0] or \
-                re.search(r'Loading\s+".+"\s+plugin', output2[0])):
+                re.search(r'Loading\s+".+"\s+plugin', output2[0]))):
             end(WARNING, "Yum output signature does not match current known "  \
                        + "format. " + support_msg)
         num_packages = 0
-        if len(output2) == 1:
+        if len(output2) == 1 and not self.no_cache_update:
             # There are no updates but we have passed
             # the loading and setting up of repositories
             pass
         else:
-            for line in output2[1].split("\n"):
+            for line in output2[-1].split("\n"):
                 if len(line.split()) > 1 and \
                    line[0:1] != " " and \
                    "Obsoleting Packages" not in line:
                     num_packages += 1
 
-        try:
```

Comment:
`num_packages` above is defined as `0` and then only incremented via `num_packages += 1` -- it will never be `<0` or not an integer, as such we can drop this check.